oauth2/oauth2-jose/src/test/java/org/springframework/security/oauth2/jwt/JwtDecodersTests.java

```diff
@@ -134,18 +180,69 @@ public void issuerWhenRequestedIssuerIsUnresponsiveThenThrowsIllegalArgumentExce
 				.isInstanceOf(IllegalArgumentException.class);
 	}
 
-	private void prepareOpenIdConfigurationResponse() {
+	@Test
+	public void issuerOAuth2WhenRequestedIssuerIsUnresponsiveThenThrowsIllegalArgumentException()
+			throws Exception {
+
+		this.server.shutdown();
+
+		assertThatCode(() -> JwtDecoders.fromOAuth2IssuerLocation("https://issuer"))
+				.isInstanceOf(IllegalArgumentException.class);
+	}
+
+	/**
+	 *
+	 * Test for compatibility of legacy application running on older version of OpenID where the server metadata
+	 * endpoint when issuer is available looks as follow "/issuer1/.well-known/openid-configruation" instead of
+	 * "/.well-knonw/openid-configuration/issuer1". This test simulates that for such cases the first requests
+	 * results in 404 however, the subsequent request results in 200 OK ensuring compatibility with legacy application.
+	 *
+	 * GET /.well-known/openid-configuration/issuer1 HTTP/1.1 and responded: HTTP/1.1 404 Client Error
+	 *
+	 * and subsequent attempt
+	 *
+	 * GET /issuer1/.well-known/openid-configuration HTTP/1.1 and responded: HTTP/1.1 200 OK
+	 * @throws Exception
+	 */
+	@Test
+	public void issuerWithPathForOIDCCompatibilityAllInformaitonThenSuccess() throws Exception {
+		prepareConfigurationResponseForCompatibility("issuer1");
+		assertThat(JwtDecoders.fromOidcIssuerLocation(this.issuer)).isNotNull();
+		assertThat(this.issuer).endsWith("/issuer1");
+	}
+
+	private void prepareConfigurationResponse() {
 		String body = String.format(DEFAULT_RESPONSE_TEMPLATE, this.issuer, this.issuer);
-		prepareOpenIdConfigurationResponse(body);
+		prepareConfigurationResponse(body);
 	}
 
-	private void prepareOpenIdConfigurationResponse(String body) {
+	private void prepareConfigurationResponse(String body) {
+		//this.issuer = createIssuerFromServer();
```

Comment:
Make sure to clean this up in your final changes

Reply:
Good catch! I took care of it 👍